src/mongo_types.ts

```diff
@@ -515,7 +515,7 @@ export type NestedPaths<Type> = Type extends
           ArrayType extends Type
           ? [Key] // we have a recursive array union
           : // child is an array, but it's not a recursive array
-            [Key, ...NestedPaths<Type[Key]>]
```

Comment:
Cool 😛, it is better than me.
It extra covers NestedPaths rootType was array type
```ts
type _ = NestedPaths<{ foo: string }[][]>;
// [number, number, "foo"]  <- before reviewed
// [] | [number] | [number, number, "foo"] <- after reviewed
```
But I think both are equivalent in case NestedPaths rootType was object